From 144712ffd0843c4ab055badf2ab1e893a221204d Mon Sep 17 00:00:00 2001
From: b prakash <bprakash@cavium.com>
Date: Fri, 10 Nov 2017 15:20:43 +0530
Subject: [PATCH 260/375] net: thunderx: Added checksum error stats to ethtool.

Added ethtool support to dump check sum description error
and checksum packet error statistics reported in sq status.

Signed-off-by: b prakash <bprakash@cavium.com>
---
 drivers/net/ethernet/cavium/thunder/nic.h           | 2 ++
 drivers/net/ethernet/cavium/thunder/nicvf_ethtool.c | 2 ++
 drivers/net/ethernet/cavium/thunder/nicvf_queues.c  | 6 ++++++
 drivers/net/ethernet/cavium/thunder/nicvf_queues.h  | 4 +++-
 4 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/thunder/nic.h b/drivers/net/ethernet/cavium/thunder/nic.h
index e69d5a15e016..17d1e25f0ba4 100644
--- a/drivers/net/ethernet/cavium/thunder/nic.h
+++ b/drivers/net/ethernet/cavium/thunder/nic.h
@@ -256,6 +256,8 @@ struct nicvf_drv_stats {
 	u64 tx_mem_fault;
 	u64 tx_csum_overlap;
 	u64 tx_csum_overflow;
+	u64 tx_csum_derr;
+	u64 tx_csum_perr;
 
 	/* driver debug stats */
 	u64 tx_tso;
diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_ethtool.c b/drivers/net/ethernet/cavium/thunder/nicvf_ethtool.c
index 6038c7eb5d55..10d267789c35 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_ethtool.c
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_ethtool.c
@@ -101,6 +101,8 @@ static const struct nicvf_stat nicvf_drv_stats[] = {
 	NICVF_DRV_STAT(tx_mem_fault),
 	NICVF_DRV_STAT(tx_csum_overlap),
 	NICVF_DRV_STAT(tx_csum_overflow),
+	NICVF_DRV_STAT(tx_csum_derr),
+	NICVF_DRV_STAT(tx_csum_perr),
 
 	NICVF_DRV_STAT(tx_tso),
 	NICVF_DRV_STAT(tx_timeout),
diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_queues.c b/drivers/net/ethernet/cavium/thunder/nicvf_queues.c
index f9c51b904fd2..3871d4a81ec9 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_queues.c
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_queues.c
@@ -1805,6 +1805,12 @@ int nicvf_check_cqe_tx_errs(struct nicvf *nic, struct cqe_send_t *cqe_tx)
 	case CQ_TX_ERROP_CK_OFLOW:
 		this_cpu_inc(nic->drv_stats->tx_csum_overflow);
 		break;
+	case CQ_TX_ERROP_CK_DERR:
+		this_cpu_inc(nic->drv_stats->tx_csum_derr);
+		break;
+	case CQ_TX_ERROP_CK_PERR:
+		this_cpu_inc(nic->drv_stats->tx_csum_perr);
+		break;
 	}
 
 	return 1;
diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_queues.h b/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
index d411a7e40dfd..23aa4e71c5fa 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
@@ -191,7 +191,9 @@ enum CQ_TX_ERROP_E {
 	CQ_TX_ERROP_MEM_FAULT = 0x87,
 	CQ_TX_ERROP_CK_OVERLAP = 0x88,
 	CQ_TX_ERROP_CK_OFLOW = 0x89,
-	CQ_TX_ERROP_ENUM_LAST = 0x8a,
+	CQ_TX_ERROP_CK_DERR = 0x8a,
+	CQ_TX_ERROP_CK_PERR = 0x8b,
+	CQ_TX_ERROP_ENUM_LAST = 0x8c,
 };
 
 enum RQ_SQ_STATS {
-- 
2.14.1

